src/NumericalAlgorithms/Interpolation/CleanUpInterpolator.hpp

```diff
@@ -63,54 +63,70 @@ struct CleanUpInterpolator {
+    // temporal_id, we will remove them.
+    bool this_temporal_id_is_done = true;
     const auto& holders =
         db::get<Tags::InterpolatedVarsHolders<Metavariables>>(box);
     tmpl::for_each<typename Metavariables::interpolation_target_tags>(
         [&](auto tag) noexcept {
           using Tag = typename decltype(tag)::type;
-          const auto& found = get<Vars::HolderTag<Tag, Metavariables>>(holders)
-                                  .times_when_data_has_been_interpolated;
-          if (found.count(time) == 0) {
-            this_time_is_done = false;
+          if constexpr (std::is_same_v<
+                            typename InterpolationTargetTag::temporal_id,
+                            typename Tag::temporal_id>) {
+            const auto& found =
+                get<Vars::HolderTag<Tag, Metavariables>>(holders)
+                    .temporal_ids_when_data_has_been_interpolated;
+            if (found.count(temporal_id) == 0) {
+              this_temporal_id_is_done = false;
+            }
           }
         });
 
-    // We don't need any more volume data for this time,
+    // We don't need any more volume data for this temporal_id,
     // so remove it.
-    if (this_time_is_done) {
-      db::mutate<Tags::VolumeVarsInfo<Metavariables>>(
+    if (this_temporal_id_is_done) {
+      db::mutate<Tags::VolumeVarsInfo<
+          Metavariables, typename InterpolationTargetTag::temporal_id>>(
           make_not_null(&box),
-          [&time](const gsl::not_null<
-                  typename Tags::VolumeVarsInfo<Metavariables>::type*>
-                      volume_vars_info) noexcept {
-            volume_vars_info->erase(time);
+          [&temporal_id](
+              const gsl::not_null<typename Tags::VolumeVarsInfo<
+                  Metavariables,
+                  typename InterpolationTargetTag::temporal_id>::type*>
+                  volume_vars_info) noexcept {
+            volume_vars_info->erase(temporal_id);
           });
 
-      // Clean up times_when_data_has_been_interpolated
+      // Clean up temporal_ids_when_data_has_been_interpolated
       db::mutate<Tags::InterpolatedVarsHolders<Metavariables>>(
           make_not_null(&box),
-          [&time](const gsl::not_null<
-                  typename Tags::InterpolatedVarsHolders<Metavariables>::type*>
-                      holders_l) noexcept {
+          [&temporal_id](
+              const gsl::not_null<
+              typename Tags::InterpolatedVarsHolders<Metavariables>::type*>
+                  holders_l) noexcept {
             tmpl::for_each<typename Metavariables::interpolation_target_tags>(
                 [&](auto tag) noexcept {
                   using Tag = typename decltype(tag)::type;
-                  get<Vars::HolderTag<Tag, Metavariables>>(*holders_l)
-                      .times_when_data_has_been_interpolated.erase(time);
+                  if constexpr (std::is_same_v<typename InterpolationTargetTag::
```

Comment:
is this necessary? (related to question about previous `if constexpr`)

Reply:
Yes -- here again, we only want to clean up the set of targets that share a temporal id type with the `InterpolationTargetTag`.